invisible_cities/reco/paolina_functions.py

```diff
@@ -321,38 +321,32 @@ def drop_voxel(voxels: Sequence[Voxel], the_vox: Voxel) -> int:
         qs  = [getattr(h, e_type) for h in the_vox.hits]
 
         if len(pos) == 0:
-            min_dist = 1e+06
-            min_v = voxels[0]
-            for v in voxels:
-                if neighbours(the_vox, v):
-                    dist = np.linalg.norm(the_vox.pos - v.pos)
-                    if dist < min_dist:
-                        min_dist = dist
-                        min_v = v
-
-            ### add voxel energy to voxel
-            min_v.E += the_vox.E
+            min_dist  = min([np.linalg.norm(the_vox.pos-v.pos) for v in voxels  if neighbours(the_vox, v)])
+            min_v     = [v for v in voxels  if (neighbours(the_vox, v) and np.isclose(np.linalg.norm(the_vox.pos-v.pos), min_dist))]
+
+            ### add dropped voxel energy to closest voxels, proportional to the  voxels energy
+            sum_en_v = sum(v.E for v in min_v)
+            for v in min_v:
+                v.E += the_vox.E/sum_en_v * v.E
 
             return 1
 
         bary_pos = np.average(pos, weights=qs, axis=0)
 
         ### find hit with minimum distance, only among neighbours
-        min_dist = 1e+06
-        min_hit = voxels[0].hits[0]
-        min_v = voxels[0]
-        for v in voxels:
-            if neighbours(the_vox, v):
-                for hh in v.hits:
-                    dist = np.linalg.norm(bary_pos - hh.pos)
-                    if dist < min_dist:
-                        min_dist = dist
-                        min_hit = hh
-                        min_v = v
-
-        ### add voxel energy to hit and to voxel, separately
-        setattr(min_hit, e_type, getattr(min_hit, e_type) + the_vox.E)
-        min_v.E += the_vox.E
+
+        min_dist    = min([np.linalg.norm(bary_pos-hh.pos) for v in voxels for hh in v.hits if neighbours(the_vox, v)])
+        min_h_v  = [(h, v) for v in voxels for h in v.hits if ((neighbours(the_vox, v) and np.isclose(np.linalg.norm(bary_pos-h.pos), min_dist)))]
```

Comment:
same comments as before apply here

Reply:
Another possibility is something like:
```
# this will order them by distance
dist_hit_vox = sorted((dist_to_bar(h), h, v) for v in voxels for h in v.hits)
closest = list(itertools.takewhile(lambda x: isclose(x, dist_hit_vox[0][0]), dist_hit_vox))
for (d, h, v) in closest:
    # do stuff
```

Not necessarily better, but something to explore.

Reply:
Discarded because the author doesn't think improves readability